File: sdkwork-backend-sdk-java/src/main/java/com/sdkwork/backend/api/MessageApi.java
package com.sdkwork.backend.api;

import com.sdkwork.backend.http.HttpClient;
import com.sdkwork.backend.model.*;
import java.util.List;
import java.util.Map;

public class MessageApi {
    private final HttpClient client;
    
    public MessageApi(HttpClient client) {
        this.client = client;
    }

    /** Update an existing chat message */
    public PlusApiResultPlusChatMessageContentVO update(PlusChatMessageContentForm body) throws Exception {
        return (PlusApiResultPlusChatMessageContentVO) client.put(ApiPaths.backendPath("/message"), body);
    }

    /** Create a new chat message */
    public PlusApiResultPlusChatMessageContentVO create(PlusChatMessageContentForm body) throws Exception {
        return (PlusApiResultPlusChatMessageContentVO) client.post(ApiPaths.backendPath("/message"), body);
    }

    /** Get chat messages by page */
    public PlusApiResultPagePlusChatMessageContentVO listByPage(QueryListForm body, Map<String, Object> params) throws Exception {
        return (PlusApiResultPagePlusChatMessageContentVO) client.post(ApiPaths.backendPath("/message/list"), body, params);
    }

    /** Get all chat messages */
    public PlusApiResultListPlusChatMessageContentVO listAllEntities(QueryListForm body) throws Exception {
        return (PlusApiResultListPlusChatMessageContentVO) client.post(ApiPaths.backendPath("/message/list/all"), body);
    }

    /** Get chat message by ID */
    public PlusApiResultPlusChatMessageContentVO getById(String id) throws Exception {
        return (PlusApiResultPlusChatMessageContentVO) client.get(ApiPaths.backendPath("/message/" + id + ""));
    }

    /** Delete a chat message */
    public PlusApiResultBoolean delete(String id) throws Exception {
        return (PlusApiResultBoolean) client.delete(ApiPaths.backendPath("/message/" + id + ""));
    }
}

File: sdkwork-backend-sdk-kotlin/src/main/kotlin/com/sdkwork/backend/api/MessageApi.kt
package com.sdkwork.backend.api

import com.sdkwork.backend.*
import com.sdkwork.backend.http.HttpClient

class MessageApi(private val client: HttpClient) {

    /** Update an existing chat message */
    suspend fun update(body: PlusChatMessageContentForm): PlusApiResultPlusChatMessageContentVO? {
        return client.put(ApiPaths.backendPath("/message"), body) as? PlusApiResultPlusChatMessageContentVO
    }

    /** Create a new chat message */
    suspend fun create(body: PlusChatMessageContentForm): PlusApiResultPlusChatMessageContentVO? {
        return client.post(ApiPaths.backendPath("/message"), body) as? PlusApiResultPlusChatMessageContentVO
    }

    /** Get chat messages by page */
    suspend fun listByPage(body: QueryListForm? = null, params: Map<String, Any>? = null): PlusApiResultPagePlusChatMessageContentVO? {
        return client.post(ApiPaths.backendPath("/message/list"), body, params) as? PlusApiResultPagePlusChatMessageContentVO
    }

    /** Get all chat messages */
    suspend fun listAllEntities(body: QueryListForm? = null): PlusApiResultListPlusChatMessageContentVO? {
        return client.post(ApiPaths.backendPath("/message/list/all"), body) as? PlusApiResultListPlusChatMessageContentVO
    }

    /** Get chat message by ID */
    suspend fun getById(id: String): PlusApiResultPlusChatMessageContentVO? {
        return client.get(ApiPaths.backendPath("/message/$id")) as? PlusApiResultPlusChatMessageContentVO
    }

    /** Delete a chat message */
    suspend fun delete(id: String): PlusApiResultBoolean? {
        return client.delete(ApiPaths.backendPath("/message/$id")) as? PlusApiResultBoolean
    }
}
